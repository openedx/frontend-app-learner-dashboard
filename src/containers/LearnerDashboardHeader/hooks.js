import React from 'react';
import { useIntl } from '@edx/frontend-platform/i18n';
import track from 'tracking';
import { StrictDict } from 'utils';
import { linkNames } from 'tracking/constants';

import getLearnerHeaderMenu from './LearnerDashboardMenu';

export const state = StrictDict({
  isOpen: (val) => React.useState(val), // eslint-disable-line
});

export const findCoursesNavClicked = (href) => track.findCourses.findCoursesClicked(href, {
  linkName: linkNames.learnerHomeNavExplore,
});

export const useLearnerDashboardHeaderMenu = ({
  courseSearchUrl, authenticatedUser, exploreCoursesClick,
}) => {
  const { formatMessage } = useIntl();
  return getLearnerHeaderMenu(formatMessage, courseSearchUrl, authenticatedUser, exploreCoursesClick);
};

export default {
  findCoursesNavClicked,
  useLearnerDashboardHeaderMenu,
};
