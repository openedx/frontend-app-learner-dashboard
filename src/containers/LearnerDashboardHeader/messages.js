import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  dashboard: {
    id: 'learnerVariantDashboard.menu.dashboard.label',
    defaultMessage: 'Dashboard',
    description: 'The text for the user menu Dashboard navigation link.',
  },
  dashboardPersonal: {
    id: 'learnerVariantDashboard.menu.dashboardPersonal.label',
    defaultMessage: 'Personal',
    description: 'Link to personal dashboard in user menu',
  },
  dashboardSwitch: {
    id: 'learnerVariantDashboard.menu.dashboardSwitch.label',
    defaultMessage: 'SWITCH DASHBOARD',
    description: 'Switch Dashboard header in the user menu',
  },
  help: {
    id: 'learnerVariantDashboard.help.label',
    defaultMessage: 'Help',
    description: 'The text for the link to the Help Center',
  },
  profile: {
    id: 'learnerVariantDashboard.menu.profile.label',
    defaultMessage: 'Profile',
    description: 'The text for the user menu Profile navigation link.',
  },
  viewPrograms: {
    id: 'learnerVariantDashboard.menu.viewPrograms.label',
    defaultMessage: 'View Programs',
    description: 'The text for the user menu View Programs navigation link.',
  },
  account: {
    id: 'learnerVariantDashboard.menu.account.label',
    defaultMessage: 'Account',
    description: 'The text for the user menu Account navigation link.',
  },
  orderHistory: {
    id: 'learnerVariantDashboard.menu.orderHistory.label',
    defaultMessage: 'Order History',
    description: 'The text for the user menu Order History navigation link.',
  },
  logout: {
    id: 'learnerVariantDashboard.menu.logout.label',
    defaultMessage: 'Logout',
    description: 'The label for the user menu Logout action.',
  },
  course: {
    id: 'learnerVariantDashboard.course',
    defaultMessage: 'Courses',
    description: 'Header link for switching to dashboard page.',
  },
  program: {
    id: 'learnerVariantDashboard.program',
    defaultMessage: 'Programs',
    description: 'Header link for switching to program page.',
  },
  discoverNew: {
    id: 'learnerVariantDashboard.discoverNew',
    defaultMessage: 'Discover New',
    description: 'Header link for switching to discover page.',
  },
  logoAltText: {
    id: 'learnerVariantDashboard.logoAltText',
    defaultMessage: 'edX, Inc. Dashboard',
    description: 'Alt text for the edX logo.',
  },
  collapseMenuOpenAltText: {
    id: 'learnerVariantDashboard.collapseMenuOpenAltText',
    defaultMessage: 'Menu',
    description: 'Alt text for the collapse menu icon when the menu is open.',
  },
  collapseMenuClosedAltText: {
    id: 'learnerVariantDashboard.collapseMenuClosedAltText',
    defaultMessage: 'Close',
    description: 'Alt text for the collapse menu icon when the menu is closed.',
  },
  career: {
    id: 'leanerDashboard.menu.career.label',
    defaultMessage: 'Career',
    description: 'The text for the user menu Career navigation link.',
  },
  newAlert: {
    id: 'header.menu.new.label',
    defaultMessage: 'New',
    description: 'The text announcing that an item in the user menu is New',
  },
});

export default messages;
