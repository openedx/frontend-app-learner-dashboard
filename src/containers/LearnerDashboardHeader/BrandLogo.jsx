import React from 'react';

import { useIntl } from '@edx/frontend-platform/i18n';

import { reduxHooks } from '@src/hooks';

import { getConfig } from '@edx/frontend-platform';
import messages from './messages';

export const BrandLogo = () => {
  const { formatMessage } = useIntl();
  const dashboard = reduxHooks.useEnterpriseDashboardData();

  return (
    <a href={dashboard?.url || '/'} className="mx-auto">
      <img
        className="logo py-3"
        src={getConfig().LOGO_URL}
        alt={formatMessage(messages.logoAltText)}
      />
    </a>
  );
};

BrandLogo.propTypes = {};

export default BrandLogo;
