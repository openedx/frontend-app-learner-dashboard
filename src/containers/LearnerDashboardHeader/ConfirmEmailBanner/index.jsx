/* eslint-disable max-len */
import React from 'react';
import {
  Button,
  Image,
  MarketingModal,
  ModalDialog,
  PageBanner,
} from '@openedx/paragon';
import { useIntl } from '@edx/frontend-platform/i18n';

import confirmEmailSVG from 'assets/confirm-email.svg';
import messages from './messages';
import './ConfirmEmailBanner.scss';
import useConfirmEmailBannerData from './hooks';

export const ConfirmEmailBanner = () => {
  const {
    isNeeded,
    showConfirmModal,
    showPageBanner,
    closePageBanner,
    closeConfirmModal,
    openConfirmModalButtonClick,
    userConfirmEmailButtonClick,
  } = useConfirmEmailBannerData();
  const { formatMessage } = useIntl();

  if (!isNeeded) { return null; }

  return (
    <>
      <PageBanner show={showPageBanner} dismissible onDismiss={closePageBanner}>
        {formatMessage(messages.confirmEmailTextReminderBanner, {
          confirmNowButton: (
            <Button
              className="confirm-email-now-button"
              variant="link"
              size="inline"
              onClick={openConfirmModalButtonClick}
            >
              {formatMessage(messages.confirmNowButton)}
            </Button>
          ),
        })}
      </PageBanner>
      <MarketingModal
        title=""
        isOpen={showConfirmModal}
        onClose={closeConfirmModal}
        hasCloseButton={false}
        heroNode={(
          <ModalDialog.Hero className="bg-gray-300">
            <Image
              className="m-auto"
              src={confirmEmailSVG}
              alt={formatMessage(messages.confirmEmailImageAlt)}
            />
          </ModalDialog.Hero>
        )}
        footerNode={(
          <Button className="mx-auto my-3" variant="danger" onClick={userConfirmEmailButtonClick}>
            {formatMessage(messages.verifiedConfirmEmailButton)}
          </Button>
        )}
      >
        <h2 className="text-center p-3 h1">{formatMessage(messages.confirmEmailModalHeader)}</h2>
        <p className="text-center">{formatMessage(messages.confirmEmailModalBody)}</p>
      </MarketingModal>
    </>
  );
};
ConfirmEmailBanner.propTypes = {};

export default ConfirmEmailBanner;
