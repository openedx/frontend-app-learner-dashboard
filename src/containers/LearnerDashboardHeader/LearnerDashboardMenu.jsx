import { getConfig } from '@edx/frontend-platform';

import urls from 'data/services/lms/urls';

import messages from './messages';

const getLearnerHeaderMenu = (
  formatMessage,
  courseSearchUrl,
  authenticatedUser,
  exploreCoursesClick,
) => ({
  mainMenu: [
    {
      type: 'item',
      href: '/',
      content: formatMessage(messages.course),
      isActive: true,
    },
    ...(getConfig().ENABLE_PROGRAMS ? [{
      type: 'item',
      href: `${urls.programsUrl()}`,
      content: formatMessage(messages.program),
    }] : []),
    ...(!getConfig().NON_BROWSABLE_COURSES ? [{
      type: 'item',
      href: `${urls.baseAppUrl(courseSearchUrl)}`,
      content: formatMessage(messages.discoverNew),
      onClick: (e) => {
        exploreCoursesClick(e);
      },
    }]
      : []),
  ],
  secondaryMenu: [
    ...(getConfig().SUPPORT_URL ? [{
      type: 'item',
      href: `${getConfig().SUPPORT_URL}`,
      content: formatMessage(messages.help),
    }] : []),
  ],
  userMenu: [
    {
      heading: '',
      items: [
        {
          type: 'item',
          href: `${getConfig().ACCOUNT_PROFILE_URL}/u/${authenticatedUser?.username}`,
          content: formatMessage(messages.profile),
        },
        {
          type: 'item',
          href: `${getConfig().ACCOUNT_SETTINGS_URL}`,
          content: formatMessage(messages.account),
        },
        ...(getConfig().ORDER_HISTORY_URL ? [{
          type: 'item',
          href: getConfig().ORDER_HISTORY_URL,
          content: formatMessage(messages.orderHistory),
        }] : []),
      ],
    },
    {
      heading: '',
      items: [
        {
          type: 'item',
          href: `${getConfig().LOGOUT_URL}`,
          content: formatMessage(messages.signOut),
        },
      ],
    },
  ],
}
);

export default getLearnerHeaderMenu;
