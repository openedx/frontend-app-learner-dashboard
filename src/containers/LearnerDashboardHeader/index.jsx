import React from 'react';

import Header from '@edx/frontend-component-header';
import MasqueradeBar from 'containers/MasqueradeBar';
import ConfirmEmailBanner from './ConfirmEmailBanner';

export const LearnerDashboardHeader = () => (
  <>
    <ConfirmEmailBanner />
    <Header />
    <MasqueradeBar />
  </>
);

LearnerDashboardHeader.propTypes = {};

export default LearnerDashboardHeader;
