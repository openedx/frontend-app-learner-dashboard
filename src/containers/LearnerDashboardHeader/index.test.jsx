import { shallow } from '@edx/react-unit-test-utils';
import Header from '@edx/frontend-component-header';

import LearnerDashboardHeader from '.';

jest.mock('containers/MasqueradeBar', () => 'MasqueradeBar');
jest.mock('./ConfirmEmailBanner', () => 'ConfirmEmailBanner');
jest.mock('@edx/frontend-component-header', () => 'Header');

describe('LearnerDashboardHeader', () => {
  test('render', () => {
    const wrapper = shallow(<LearnerDashboardHeader />);
    expect(wrapper.snapshot).toMatchSnapshot();
    expect(wrapper.instance.findByType('ConfirmEmailBanner')).toHaveLength(1);
    expect(wrapper.instance.findByType('MasqueradeBar')).toHaveLength(1);
    expect(wrapper.instance.findByType(Header)).toHaveLength(1);
  });
});
