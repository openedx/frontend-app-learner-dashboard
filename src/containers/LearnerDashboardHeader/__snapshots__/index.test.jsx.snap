// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearnerDashboardHeader render 1`] = `
<Fragment>
  <ConfirmEmailBanner />
  <Header />
  <MasqueradeBar />
</Fragment>
`;
